File: algo/code/04/js/04remove_elem.js
/* 删除元素 */


function removeElem(nums, index) {
    for (let i = index; i < nums.length - 1; i++) {
        nums[i] = nums[i+1]
    }
    return nums
}

// 调用函数
nums = [1, 3, 2, 5, 4]
console.log(nums) // [1, 3, 2, 5, 4]
console.log(removeElem(nums, 3)) // [1, 3, 2, 4, 4]

File: algo/code/04/c/04removeElem.c
/* 删除元素 */

/* 删除索引 index 处的元素 */
// 注意：stdio.h 占用了 remove 关键词
void removeItem(int *nums, int size, int index) {
    // 把索引 index 之后的所有元素向前移动一位
    for (int i = index; i < size - 1; i++) {
        nums[i] = nums[i + 1];
    }
}